### PR TITLE
Refactor JUnit runner.

### PR DESCRIPTION
**Pull request by arturdryomov**
Just inlining and renamings to make more sense with JUnit in mind. If Spek implements some JUnit interface — it is prepended with `Spek` prefix.

### CODE REVIEW

File: spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    override fun setParent(parent: TestDescriptor?) {
+        // Called only when adding as a child of the engine's descriptor.
+        // Will be null only if it's not the root scope.
+        if (parent != null && parent is SpekEngineDescriptor) {
+            engineDescriptor = parent
+        }
+    }
+
+    override fun getParent(): Optional<TestDescriptor> {
+        // TODO: replace with just `scope.parent`?
```
Comment (arturdryomov) [inline spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt]: @raniejade, please take a look at this. Seems like the whole expression can be simplified to `scope.parent as ScopeImpl?`.

**Comment by arturdryomov** [comment]
@raniejade, 🏓

File: spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt (hunk 1 above)
Comment (raniejade) [reply spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt]: Go ahead 👍.

**Review by raniejade — APPROVED** [review]

File: spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt (hunk 1 above)
Comment (arturdryomov) [reply spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt]: @raniejade, done, thanks!

FYI — I don’t have rights to merge PRs, so please do it yourself if everything else looks fine.

**Pull request merged**